stripe/src/main/java/com/stripe/android/paymentsheet/model/ClientSecret.kt

```diff
@@ -0,0 +1,23 @@
+package com.stripe.android.paymentsheet.model
+
+import android.os.Parcelable
+import com.stripe.android.model.PaymentIntent
+import com.stripe.android.model.SetupIntent
+import kotlinx.parcelize.Parcelize
+
+/**
+ * Represents the client secret for a [SetupIntent] or [PaymentIntent]
+ */
+internal sealed class ClientSecret : Parcelable {
+    abstract val value: String
+
+    @Parcelize
+    data class PaymentIntentClientSecret(
+        override val value: String
+    ) : ClientSecret()
+
+    @Parcelize
+    data class SetupIntentClientSecret(
+        override val value: String
+    ) : ClientSecret()
+}
```

Comment:
Can we change to:

```
internal sealed class ClientSecret : Parcelable {
    abstract val value: String
}

@Parcelize
internal data class PaymentIntentClientSecret(
    override val value: String
) : ClientSecret()

@Parcelize
internal data class SetupIntentClientSecret(
    override val value: String
) : ClientSecret()
```

Reply:
Also doesn't have to be here, but at some point would be nice to consolidate this and `PaymentIntent.ClientSecret`.

Reply:
Done. Will make `PaymentIntent.ClientSecret` and `SetupIntent.ClientSecret` methods, they're only used to extract the id from the client secret.